From: Miaohe Lin <linmiaohe@huawei.com>
Subject: mm/migration: fix the confusing PageTransHuge check

prep_transhuge_page should be called when PageTransHuge(page) is true. 
The newly allocated new_page is not yet PageTransHuge though it could pass
the check as PageTransHuge only checks PageHead now.

Link: https://lkml.kernel.org/r/20220318111709.60311-5-linmiaohe@huawei.com
Signed-off-by: Miaohe Lin <linmiaohe@huawei.com>
Reviewed-by: Muchun Song <songmuchun@bytedance.com>
Cc: Alistair Popple <apopple@nvidia.com>
Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: "Huang, Ying" <ying.huang@intel.com>
Cc: Zi Yan <ziy@nvidia.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/migrate.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/migrate.c~mm-migration-fix-the-confusing-pagetranshuge-check
+++ a/mm/migrate.c
@@ -1565,7 +1565,7 @@ struct page *alloc_migration_target(stru
 
 	new_page = __alloc_pages(gfp_mask, order, nid, mtc->nmask);
 
-	if (new_page && PageTransHuge(new_page))
+	if (new_page && PageTransHuge(page))
 		prep_transhuge_page(new_page);
 
 	return new_page;
_
